bmwqemu.pm

```diff
@@ -317,6 +321,39 @@ sub TIEHASH ($class, %args) {
 sub STORE ($self, $key, $val) {
     croak("Settings key '$key' is invalid (check your test settings)") unless $key =~ m/^(?:[A-Z0-9_]+)\z/;
     $self->{data}->{$key} = $val;
+    my @stack;
+    for (0 .. 9) {
+        my @caller = caller($_);
+        last unless @caller;
+        my ($package, $file, $line, $sub) = @caller;
+        if ($caller[3] eq 'bmwqemu::load_vars') {
+            $self->{trace}->{load_vars}->{$key} = $val;
+            return;
+        }
+        next if ($package eq 'testapi');
+        last if ($package eq 'basetest' and $sub eq '(eval)');
+        last if $package eq 'main';
+        push @stack, [$package, $line, $sub];
+    }
+    $self->{trace}->{override}->{$key} = {
+        callstack => \@stack,
+        value => $val,
+    };
+}
+
+sub debug_vars ($self) {
+    use YAML::PP;
+    use YAML::PP::Common qw/ PRESERVE_FLOW_STYLE YAML_FLOW_SEQUENCE_STYLE /;
+    my $yp = YAML::PP->new(preserve => PRESERVE_FLOW_STYLE);
+    my $trace = $self->{trace};
+    for my $key (keys %{$trace->{override}}) {
+        my $callstack = $trace->{override}->{$key}->{callstack};
+        for my $row (@$callstack) {
+            $row = $yp->preserved_sequence($row, style => YAML_FLOW_SEQUENCE_STYLE);
+        }
+    }
```

Comment:
If anyone wonders - this is just for making the output less noisy by using flow style `[foo, bar]` instead of
```
- foo
- bar
```